### PR TITLE
[#308] Retry when didn't find instance for a first time.

### PR DESCRIPTION
**Pull request by cameleeck**

### CODE REVIEW

File: stub-runner/stub-runner/README.md [hunk 1]
```diff
@@ -70,6 +75,17 @@ When you want to register your stubs against embedded Zookeeper instance it's en
 that will start a testing Zookeeper instance on port 1234
 
 
+##### Configuring wait time for registering services
+
+There is a option to wait given amount of time waiting for service to be registered. 
+You can do that using:
```
Comment (marcingrzejszczak) [inline stub-runner/stub-runner/README.md]: ...using an application parameter called:

File: stub-runner/stub-runner/src/main/groovy/com/ofg/stub/CollaboratorsPathResolver.groovy [hunk 2]
```diff
@@ -28,33 +33,120 @@ class CollaboratorsPathResolver {
...
+        serviceCache.addListener(listener)
+        instance = listener.get(waitTimeout, TimeUnit.SECONDS)
+        return instance
     }
 
-    @InheritConstructors
-    static class LookedUpServiceUnavailableException extends RuntimeException {
+    private static class TimeoutServiceCacheListener implements ServiceCacheListener, Future<ServiceInstance<Void>> {
+
+        ServiceProvider serviceProvider
```
Comment (marcingrzejszczak) [inline stub-runner/stub-runner/src/main/groovy/com/ofg/stub/CollaboratorsPathResolver.groovy]: private final?

File: stub-runner/stub-runner/src/main/groovy/com/ofg/stub/StubRunnerMain.groovy [hunk 3]
```diff
@@ -88,6 +94,7 @@ class StubRunnerMain {
         if (isNotBlank(arguments.stubRunnerOptions.zookeeperConnectString)) {
             return new ZookeeperServer(arguments.stubRunnerOptions.zookeeperConnectString)
         } else if (arguments.stubRunnerOptions.zookeeperPort) {
+            arguments.stubRunnerOptions.zookeeperConnectString = "localhost:" + arguments.stubRunnerOptions.zookeeperPort
```
Comment (marcingrzejszczak) [inline stub-runner/stub-runner/src/main/groovy/com/ofg/stub/StubRunnerMain.groovy]: use GString

**Comment by cameleeck** [comment]
Merged.

**Pull request closed without merging**